game/scripts/vscripts/Sections/Chapter4/SectionOutpost.ts

```diff
@@ -161,23 +161,30 @@ function onStart(complete: () => void) {
 
             tg.immediate(context => {
                 const riki = getOrError(context[CustomNpcKeys.Riki] as CDOTA_BaseNPC | undefined);
-                riki.RemoveModifierByName(modifier_abs_no_damage.name)
+                riki.RemoveModifierByName(modifier_abs_no_damage.name);
                 riki.SetAttackCapability(DOTAUnitAttackCapability_t.DOTA_UNIT_CAP_MELEE_ATTACK);
-                riki.MoveToTargetToAttack(playerHero);
             }),
 
             tg.forkAny([
                 tg.seq([
                     tg.audioDialog(LocalizationKey.Script_4_Outpost_9, LocalizationKey.Script_4_Outpost_9, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
                     tg.neverComplete()
                 ]),
-                tg.seq([
-                    tg.completeOnCheck(context => {
-                        const riki = getOrError(context[CustomNpcKeys.Riki] as CDOTA_BaseNPC | undefined);
-                        return !IsValidEntity(riki) || !riki.IsAlive();
-                    }, 1),
-                ])
+                tg.loop(context => {
+                    const riki = getOrError(context[CustomNpcKeys.Riki] as CDOTA_BaseNPC | undefined);
+                    return riki.IsAlive();
```

Comment:
Maybe instead of getOrError just return false if wiki is invalid (perhaps we removed him from the game for example, or for some reason he decayed withi nthe 2s wait)